week3/project/ecommerce3/src/App.js

```diff
@@ -0,0 +1,45 @@
+import { Routes, Route, Navigate } from "react-router-dom";
+import "./App.css";
+import HeaderContainer from "./components/HeaderContainer";
+import FavoriteItemsPage from "./components/pages/FavoriteItemsPage";
+import ProductDetailsPage from "./components/pages/ProductDetailsPage";
+import ProductsListPage from "./components/pages/ProductsListPage";
+import { FavoriteContextProvider } from "./context/FavoriteContext";
+
+function App() {
+  return (
+    <div className="App">
+      <HeaderContainer />
+      <Routes>
+        <Route path="/" element={<Navigate replace to="/products" />} />
+        <Route
+          path="/products"
+          element={
+            <FavoriteContextProvider>
```

Comment:
you almost wrap everything with the context .. so you can just wrap the whole routers with the context
📝advise📝